### ISSUE TITLE
train_mode.py

### ISSUE THREAD

**Issue by Nikitabhatkudav** [Nikitabhatkudav/Stueden_Performance_Prediction_System_DSLC#5]
import os
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import accuracy_score
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Input
from tensorflow.keras.callbacks import EarlyStopping
import joblib

DATA_PATH = os.path.join('data', 'students.csv')
MODEL_PATH = 'student_model.h5'
SCALER_PATH = 'scaler.pkl'

def load_data(path: str):
    df = pd.read_csv(path)
    X = df[['study_hours', 'attendance', 'previous_marks']].astype(float).values
    y = df['pass'].astype(int).values
    return X, y

def build_model(input_dim: int) -> Sequential:
    model = Sequential([
        Input(shape=(input_dim,)),
        Dense(8, activation='relu'),
        Dense(4, activation='relu'),
        Dense(1, activation='sigmoid')
    ])
    model.compile(optimizer='adam', loss='binary_crossentropy', metrics=['accuracy'])
    return model

def train_and_save():
    X, y = load_data(DATA_PATH)

    scaler = StandardScaler()
    X_scaled = scaler.fit_transform(X)

    X_train, X_val, y_train, y_val = train_test_split(
        X_scaled, y, test_size=0.2, random_state=42, stratify=y
    )

    model = build_model(input_dim=X_train.shape[1])

    es = EarlyStopping(monitor='val_loss', patience=20, restore_best_weights=True)

    model.fit(
        X_train,
        y_train,
        validation_data=(X_val, y_val),
        epochs=300,
        batch_size=4,
        verbose=0,
        callbacks=[es]
    )

    y_val_pred = (model.predict(X_val, verbose=0) > 0.5).astype(int).ravel()
    acc = accuracy_score(y_val, y_val_pred)
    print(f'Validation Accuracy: {acc:.2f}')

    model.save(MODEL_PATH)
    joblib.dump(scaler, SCALER_PATH)
    print(f'Saved model to {MODEL_PATH} and scaler to {SCALER_PATH}')

if __name__ == '__main__':
    os.makedirs('data', exist_ok=True)
    train_and_save()
